bootstrap/kubeadm/api/v1alpha4/condition_consts.go

```diff
@@ -49,6 +49,11 @@ const (
 	// an error while generating a data secret; those kind of errors are usually due to misconfigurations
 	// and user intervention is required to get them fixed.
 	DataSecretGenerationFailedReason = "DataSecretGenerationFailed"
+
+	// IgnitionHasWarningsReason (Severity=Warning) documents a KubeadmConfig controller detecting
+	// warnings while generating Ignition configuration; those kind of warnings are not harmful, but users
+	// should be able to find out about them.
+	IgnitionHasWarningsReason = "IgnitionHasWarnings"
```

Comment:
Just to double check the intent here.

When there will be warning, you do expect that
- `DataSecretAvailable` condition on `KubeadmConfig` will be set to `False`, Reason `IgnitionHasWarnings`
- As a consequence
  - `Ready` condition on `KubeadmConfig` will be set to `False`
  - `BootstrapReady` condition on the corresponding `Machine` will be report `False`
  - `Ready` condition on the corresponding `Machine` will be set to `False`
  
Also, see comment below about how this reason is applied.

Reply:
Hmm, I think this code should be removed and we should do warning handling via webhook as we discussed in other thread, as a separate issue, as it's a bit out of scope for this PR.

Do you think it's fine to hide the warnings for now then? if user configures strict mode, warnings will be returned as errors.

Reply:
Ok for me.

Reply:
I'll resolve. Warnings will be handled in https://github.com/kubernetes-sigs/cluster-api/issues/4279.